README.md

```diff
@@ -1,173 +1,175 @@
+
+-   **`"Challenged"`**: someone challenged the Erdstall operator, who must now respond with the challenged balance proof.
+
+-   **`"ChallengeResponded"`**: the Erdstall operator successfully responed with the correct balance proof.
+
+-   **`"TokenTypeRegistered"`**: a new token type (such as ETH, ERC20, ERC721, etc.) has been registered with Erdstall on-chain.
+
+-   **`"TokenRegistered"`**: a specific token contract has been registered on-chain.
+</details>
+
+## Development
 
 ### TypedJSON
 
 The Erdstall SDK internally uses [`TypedJSON`](https://github.com/JohnWeisz/TypedJSON) to define class JSON serialization.
 If you want to use TypedJSON in your project for class serialization, possibly using some Erdstall classes as field types, you _must_ use the re-exported module `typedjson` from `@polycrypt/erdstall/export/typedjson` like
+
 ```ts
-import { jsonObject, jsonMember, TypedJSON } from "@polycrypt/erdstall/export/typedjson";
+import {
+	jsonObject,
+	jsonMember,
+	TypedJSON,
+} from "@polycrypt/erdstall/export/typedjson";
 ```
 
 The reason is that TypedJSON internally registers class serializations in static members of the `TypedJSON` class.
 If you use your own version of `typedjson` from your `node_modules`, these modules will not share state and serialization will not work.
 
-## Using the Erdstall client
+#### `bigint` members
 
-The Erdstall client mainly offers these operations:
+If you use TypedJSON to serialize your own classes and they have a field of type `bigint`, it is best to use the annotation `@jsonBigIntMember()` from `@polycrypt/erdstall/export/typedjson`, which defines a base-10 string serialization format.
 
-**Deposits and withdrawals**&emsp;<!--
--->`Erdstall.deposit(erdstall.ledger.Assets)` creates a bunch of Ethereum transactions that need to be sent on-chain to deposit the requested amount of assets into Erdstall.
-Once the transactions are confirmed on-chain, the funds will be made available in Erdstall.
-To withdraw your funds, `Erdstall.leave()` can be used.
-It requests that the user exits, then, after a balance proof is issued, returns a bunch of Ethereum transactions that need to be sent on-chain immediately.
-These transactions will withdraw the user's funds from Erdstall back into his normal wallet.
+### Contract bindings
 
-**Subscriptions**&emsp;<!--
--->Subscribe to events using `Erdstall.subscribeTXs(filter: erdstall.ledger.Address?)` and `Erdstall.subscribeBalanceProofs(filter: erdstall.ledger.Address?)`.
-Both of these calls will listen for their respective events for either all accounts, or only for the account passed to the call.
-Each call also has a corresponding `unsubscribe` version.
-The current balance of any account can be queried using `Erdstall.getAccount(erdstall.ledger.Address)`.
+Contract bindings can be generated from the [`erdstall-contracts`]("https://github.com/perun-network/erdstall-contracts") repository.
```

Comment:
The hyperlink does not work for me, maybe with a `<code/>` block or the link without quotation marks?